WireDomain/Sources/WireDomain/Repositories/FeatureConfigRepository.swift

```diff
@@ -0,0 +1,363 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+import Combine
+import WireAPI
+import WireDataModel
+
+/// Facilitates access to feature configs related domain objects.
+
+protocol FeatureConfigRepositoryProtocol {
+
+    /// Pulls feature configs from the server and stores them locally.
+    ///
+    /// - Returns: A list of `FeatureState`.
+
+    func pullFeatureConfigs() async throws -> [FeatureState]
+
+    /// Observes feature states.
+    ///
+    /// Each time `pullFeatureConfigs()` is called, a feature config is
+    /// stored locally and a new `FeatureState` value is produced by the publisher.
+    /// It allows the user to be notified of any feature changes over time.
+    ///
+    /// ⚠️ Use this method before calling `pullFeatureConfigs` to receive all emitted values.
```

Comment:
I recommend using `- Warning:` below the `- Returns:`:

```swift
/// - Warning: Use this method before calling `pullFeatureConfigs` to receive all emitted values.
```

That way we see a nice warning in docs:
<img width="444" alt="Screenshot 2024-08-30 at 15 20 23" src="https://github.com/user-attachments/assets/8529c43d-41e1-4efc-8284-4de27077b97e">

Reply:
Nice, I didn't know about this one. thx